lib/operations/operation.js

```diff
@@ -44,6 +46,14 @@ class OperationBase {
   }
 }
 
+class OperationBaseV2 extends OperationBase {
+  constructor(parent, options) {
+    super(options);
+
+    this.readPreference = parent ? parent.readPreference : ReadPreference.primary;
```

Comment:
It seems like this is very similar to `CommandOperation` where the constructor does `this.options.readPreference = resolveReadPreference(collection, options);` What is the benefit of creating `OperationBaseV2` for this behavior?

Reply:
you're right that eventually that these methods should eventually just use the `CommandOperation`, however, that operation currently has some assumptions about how the operation is being executed (using `executeCommand` with something that looks like a `Db` object, switching behavior for write operations, etc). I was trying to minimize the impact on the rest of the codebase in this initial PR. 

I'll go ahead and make a `CommandOperationV2` instead that we can start iterating on, and migrating towards. I think that will probably address most of your questions above as well.